assets/app/scripts/controllers/create.js

```diff
@@ -94,6 +107,213 @@ angular.module('openshiftConsole')
+        JSON.parse($scope.editor.content);
+        $scope.editor.session.setMode("ace/mode/json");
+      } catch (e) {
+        try {
+          YAML.parse($scope.editor.content);
+          $scope.editor.session.setMode("ace/mode/yaml");
+        } catch (e) {
+          return;
+        }
+      }
+    };
+
+    $scope.create = function() {
+      delete $scope.alerts['create'];
+      delete $scope.alerts['parsing'];
+      var resource;
+
+      // Trying to auto-detect what format the input is in. Since parsing JSON throws only SyntexError
+      // exception if the string to parse is not valid JSON, it is tried first and then the YAML parser
+      // is trying to parse the string. If that fails it will print the reason. In case the real reason
+      // is JSON related the printed reason will be "Reason: Unable to parse", in case of YAML related
+      // reason the true reason will be printed, since YAML parser throws an error object with needed
+      // data.
+      try {
+        resource = JSON.parse($scope.editor.content);
+      } catch (e) {
+        try {
+          resource = YAML.parse($scope.editor.content);
+        } catch (e) {
+          $scope.alerts['parsing'] = {
+            type: "error",
+            message: "An error occurred during parsing editors content.",
+            details: "Reason: " + e.message
+          };
+          return;
+        }
+      }
+
+      $scope.resourceKind = resource.kind;
+      
+      if ($scope.resourceKind.endsWith("List")) {
+        $scope.resourceList = resource.items;
+      } else {
+        $scope.resourceList = [resource];
+        $scope.resourceName = resource.metadata.name;
+      }
+
+      $scope.updateResources = [];
+      $scope.createResources = [];
+
+      var itemsProcessed = 0;
+      angular.forEach($scope.resourceList, function(item) {
+        var itemKind = item.kind;
+        var itemName = item.metadata.name;
+
+        if (item.metadata.namespace && item.metadata.namespace !== $scope.projectName) {
+          $scope.alerts['create'] = {
+            type: "error",
+            message: "Can't create component in different projects.",
+            details: itemKind + " " + itemName + " can't be created in namespace " + item.metadata.namespace
+          };
+          return;
+        }
+
+        // Check if the resource already exists. If it does, replace it spec with the new one.
+        DataService.get(APIService.kindToResource(itemKind), itemName, {namespace: $scope.projectName}, {errorNotification: false}).then(
+          // resource does exist
+          function(resource) {
+            resource.spec = item.spec;
+            $scope.updateResources.push(resource);
+            itemsProcessed++;
+            if (itemsProcessed === $scope.resourceList.length) {
+              if (!_.isEmpty($scope.updateResources)) {
+                openModal();
+              } else {
+                createAndUpdate();
+              }
+            }
```

Comment:
You should be able to move this code after the `angular.forEach`, then you don't need to track `itemsProcessed`. If you want to track if there were errors, you could set a flag that you check later.